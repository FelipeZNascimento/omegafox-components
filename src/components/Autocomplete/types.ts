export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
