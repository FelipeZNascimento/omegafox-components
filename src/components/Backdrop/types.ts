export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
