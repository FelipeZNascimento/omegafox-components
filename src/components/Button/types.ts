export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
