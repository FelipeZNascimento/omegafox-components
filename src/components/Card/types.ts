export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
