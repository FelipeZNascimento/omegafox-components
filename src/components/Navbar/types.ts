export type TNavbarButton = {
  id: number;
  isDisabled?: boolean;
  text: string;
  url: string;
  withCredentials?: boolean;
  renderingFunction?: () => JSX.Element;
};

export interface INavbarProps {
  isSticky?: boolean;
  logo: string;
  navbarLeft?: TNavbarButton[];
  navbarRight?: TNavbarButton[];
  platform: 'copa' | 'nfl';
  selectedId: number;
  onClick: (navbarButton: TNavbarButton) => void;
}
