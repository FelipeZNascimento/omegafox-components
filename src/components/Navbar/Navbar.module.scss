@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.navContainer {
  @include padded(vertical, s);
  @include padded(horizontal, s);
  @include shadowed($color-black, 0.2);

  height: 64px;
  overflow: hidden;
  z-index: $always-on-top;

  &Sticky {
    align-self: flex-start;
    position: -webkit-sticky;
    position: sticky;
    top: 0;
  }

  &Copa {
    @include bg-colored($color-grey1);
  }

  &NFL {
    @include bg-colored($color-navy);
  }
}

.container {
  @include font-sized(l2);
  @include marged(all, zero);

  display: flex;
  height: 100%;
}

.logo {
  @include padded(horizontal, s);

  flex: 0;
  height: 100%;
  width: auto;
}

.buttonContainer {
  @include padded(horizontal, s);

  display: flex;
  flex: 1;

  // button {
  //   @include padded(horizontal, m);
  //   @include marged(horizontal, s3);
  //   // @include bg-colored($color-grey1);

  //   border-radius: 16px;
  //   height: 100%;
  //   transition: all $transition-time;
  // }

  button,
  input[type="submit"],
  input[type="reset"] {
    // color: inherit;
    border: none;
    cursor: pointer;
    outline: inherit;
  }

  &Left {
    @include flex-aligned(flex-start, center);
  }

  &Right {
    @include flex-aligned(flex-end, center);
  }

  &Copa {
    @extend .buttonContainer;

    .navButton {
      @include font-colored($color-grey4);
      @include padded(horizontal, m);
      @include marged(horizontal, s3);

      font: inherit;
      border-radius: 16px;
      height: 100%;
      transition: all $transition-time;

      &Selected {
        @include font-colored($color-qatar-red);
      }

      &:hover {
        @include bg-colored($color-grey2);

        transition: all $transition-time;
      }
    }
  }
}
