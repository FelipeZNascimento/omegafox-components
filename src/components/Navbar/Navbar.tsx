import { INavbarProps, TNavbarButton } from '../types';
import { NavbarButton } from '../index';
import classNames from 'classnames';
import styles from './Navbar.module.scss';

export const Navbar = ({
  isRounded = false,
  isSlim = false,
  isSticky = true,
  sameSizeButtons = false,
  navbarLeft = [],
  navbarRight = [],
  theme,
  selectedId,
  shadow = 'black',
  logo = null,
  onClick
}: INavbarProps) => {
  const navContainerClass = classNames(styles.navContainer, {
    [styles.navContainerSlim]: isSlim,
    [styles.navContainerTall]: !isSlim,
    [styles.navContainerSticky]: isSticky,
    [styles.navContainerRounded]: isRounded,
    [styles.navContainerBlackShadow]: shadow === 'black',
    [styles.navContainerWhiteShadow]: shadow === 'white',
    [styles.navContainerCopa]: theme === 'copa',
    [styles.navContainerValeo]: theme === 'valeo'
    // [styles.navContainerNfl]: theme === 'Nfl'
  });

  const renderButtons = (navbarButtons: TNavbarButton[]) => {
    return navbarButtons.map((item) => (
      <NavbarButton
        key={item.id}
        button={item}
        isFlexed={sameSizeButtons}
        isSelected={selectedId === item.id}
        theme={theme}
        onClick={onClick}
      />
    ));
  };

  const renderLeftSide = () => {
    if (navbarLeft.length === 0) {
      return null;
    }

    return (
      <div className={styles.buttonContainerLeft}>
        {renderButtons(navbarLeft)}
      </div>
    );
  };

  const renderRightSide = () => {
    if (navbarRight.length === 0) {
      return null;
    }

    return (
      <div className={styles.buttonContainerRight}>
        {renderButtons(navbarRight)}
      </div>
    );
  };

  return (
    <div className={navContainerClass}>
      <nav className={styles.container}>
        {logo && <img alt="Logo" className={styles.logo} src={logo} />}
        {renderLeftSide()}
        {renderRightSide()}
      </nav>
    </div>
  );
};
