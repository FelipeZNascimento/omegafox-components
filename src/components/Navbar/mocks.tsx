import { TNavbarButton } from './types';

export const navbarButtons: TNavbarButton[] = [
  {
    id: 0,
    text: 'InÃ­cio',
    url: ''
  },
  {
    id: 1,
    text: 'Resultados',
    url: ''
  },
  {
    id: 2,
    text: 'Apostar',
    url: ''
  },
  {
    id: 3,
    text: 'Regras',
    url: ''
  },
  {
    id: 4,
    text: 'PrÃªmios',
    url: ''
  },
  {
    id: 5,
    text: 'Extras',
    url: ''
  },
  {
    id: 6,
    text: '',
    url: '',
    renderingFunction: () => {
      return <a href="#">link</a>;
    }
  }
];
