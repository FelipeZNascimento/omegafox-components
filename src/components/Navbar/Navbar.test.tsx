import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';

import { Navbar } from './Navbar';
import { NavbarProps } from './types';

describe('NavbarComponent', () => {
  const renderComponent = ({}: NavbarProps) => render(<Navbar />);

  it('should render', () => {
    const { container } = renderComponent({});

    expect(container).toMatchSnapshot();
  });
});
