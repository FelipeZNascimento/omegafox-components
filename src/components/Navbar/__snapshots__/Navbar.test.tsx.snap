// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent should render 1`] = `
<div>
  <div
    class="navContainer navContainerSticky navContainerCopa"
  >
    <nav
      class="container"
    >
      <img
        alt="Logo"
        class="logo"
        src="any"
      />
      <div
        class="buttonContainerLeft buttonContainerCopa"
      >
        <button
          class="navButton navButtonSelected"
        >
          Início
        </button>
        <button
          class="navButton"
        >
          Resultados
        </button>
        <button
          class="navButton"
        >
          Apostar
        </button>
        <button
          class="navButton"
        >
          Regras
        </button>
        <button
          class="navButton"
        >
          Prêmios
        </button>
        <button
          class="navButton"
        >
          Extras
        </button>
      </div>
    </nav>
  </div>
</div>
`;
